### DISCUSSION TITLE
Add creation and last update dates to authorization models

### DISCUSSION THREAD

**Discussion by geoffroybraun** [openfga/community#159] [Ideas]
So far, only stores have both creation and last update dates. Unfortunatly, regarding authorization models, it is impossible to get a specific version without knowing its ID, as the `list authorization models` endpoint does not indicate how models are ordered before being retrieved.

A successful but quite easy-to-implement solution would retrieve an authorization model creation date aside with its last update one, which could be the same date as the creation date if no update has been performed yet. That way, we could list authorization models and know which one is the latest.

Please let know if more explanations have to be provided, cheers!

**Comment by rhamzeh** [comment]
Authorization Models in OpenFGA are immutable - so you cannot update them, just create new authorization models with new IDs.

Re: date created, the authorization model id is a [ulid](https://github.com/ulid/spec) which contains the date.

For example, in JS you can use the [ulid](https://github.com/ulid/javascript) library to get the time:
```
import ulid = require('ulid');
const time = ulid.decodeTime(id)
```

**Comment by aaguiarz** [comment]
The CLI also provides an easy way to get the latest authorization model

```
$ fga model get --field id --format json
{
  "id":"01HQWXWDHK9RGR1J2G3EJ01YWV"
}
```
